pulsar-client/src/main/java/org/apache/pulsar/client/impl/TransactionMetaStoreHandler.java

```diff
@@ -117,7 +117,8 @@ public void start() {
     public void connectionFailed(PulsarClientException exception) {
         LOG.error("Transaction meta handler with transaction coordinator id {} connection failed.",
             transactionCoordinatorId, exception);
-        if (!this.connectFuture.isDone()) {
+        //If the exception is a retry exception, we should not complete exceptionally at this time.
+        if (!this.connectFuture.isDone() || !isRetryException(exception)) {
```

Comment:
How the connect request can retry?

Reply:
It will do a retry operation after getting an exception
https://github.com/apache/pulsar/blob/c17f355428e544c0d14fe24848bfb9fed5e3ef48/pulsar-client/src/main/java/org/apache/pulsar/client/impl/TransactionMetaStoreHandler.java#L150-L60